polkadot/node/network/collator-protocol/src/collator_side/mod.rs

```diff
@@ -686,22 +699,40 @@ async fn connect_to_validators<Context>(
 		Vec::new()
 	};
 
-	gum::trace!(
-		target: LOG_TARGET,
-		?cores_assigned,
-		"Sending connection request to validators: {:?}",
-		validator_ids,
-	);
-
 	// ignore address resolution failure
 	// will reissue a new request on new collation
 	let (failed, _) = oneshot::channel();
-	ctx.send_message(NetworkBridgeTxMessage::ConnectToValidators {
-		validator_ids,
-		peer_set: PeerSet::Collation,
-		failed,
-	})
-	.await;
+
+	let msg = if connect {
```

Comment:
Looking at the implementation of [`list_of_backing_validators_in_view` ](https://github.com/paritytech/polkadot-sdk/blob/9b7b37f904d80b4c71feb592d48ee4add6c10d15/polkadot/node/network/collator-protocol/src/collator_side/mod.rs#L672-L672) its not filtering the validators at all. Why do we want to connect to everyone? We should filter by the groups with cores that we are scheduled on or not?

Reply:
Ah, so it contained only the ones belonging to our para in the first place, got it!